selfdrive/athena/athenad.py

```diff
@@ -244,6 +254,82 @@ def b64jpeg(x):
     raise Exception("not available while camerad is started")
 
 
+def get_logs_to_send_sorted(curr_time, last_scan):
+  # TODO: scan once then use inotify to detect file creation/deletion
+  logs = []
+  for log_entry in os.listdir(SWAGLOG_DIR):
+    log_path = os.path.join(SWAGLOG_DIR, log_entry)
+    # file without an extension is always the active log file
+    if log_entry == "swaglog":
+      continue
+    try:
+      time_sent = int.from_bytes(getxattr(log_path, LOG_ATTR_NAME), sys.byteorder)
+    except (ValueError, TypeError):
+      time_sent = 0
+    # assume send failed and we lost the response if sent more than one hour ago
+    if not time_sent or curr_time - time_sent > 3600:
```

Comment:
This also needs to take into account the system time starting on midnight 2020/04/01 and switching to the current time after boardd is started.

Reply:
I think this is OK since there is only ever a single log file in flight at a time, and so worst case we send one file twice which shouldn't cause any problems, what do you think?

Reply:
I would just use sec_since_boot for this time, since it doesn't matter what it is as long as you can measure durations.

Reply:
it does matter, because if it reboots with a high monotonic time it will take forever to consider it eligible to re-send, right?